azkaban-common/src/main/java/azkaban/executor/ExecutionControllerUtils.java

```diff
@@ -230,7 +230,12 @@ public static void failEverything(final ExecutableFlow exFlow) {
           continue;
           // case UNKNOWN:
         case READY:
-          node.setStatus(Status.KILLING);
+          // if flow status is EXECUTION_STOPPED due to e.g. pod failure, set sub node to KILLED
+          if (exFlow.getStatus()==Status.EXECUTION_STOPPED) {
+            node.setStatus(Status.KILLED);
```

Comment:
Just to confirm, this will take care of all the nodes in graph. right?

Reply:
+1

Reply:
No, the current code is only taking care of first level of nodes, ignoring the sub-nodes within an embedded flow. I'll refactor the code to traverse the DAG.